nuxeo-core/nuxeo-core-storage-mongodb/src/main/java/org/nuxeo/ecm/core/storage/mongodb/MongoDBChecker.java

```diff
@@ -57,28 +65,34 @@ public boolean accepts(ConfigurationGenerator cg) {
 
     @Override
     public void check(ConfigurationGenerator cg) throws ConfigurationException {
-        MongoClient ret = null;
-        String serverName = cg.getUserConfig().getProperty(ConfigurationGenerator.PARAM_MONGODB_SERVER);
-        String dbName = cg.getUserConfig().getProperty(ConfigurationGenerator.PARAM_MONGODB_NAME);
-
-        MongoClientOptions.Builder optionsBuilder = MongoClientOptions.builder()
-                                                                      .serverSelectionTimeout(
-                                                                              (int) TimeUnit.SECONDS.toMillis(getCheckTimeoutInSeconds(cg)))
-                                                                      .description("Nuxeo DB Check");
-        if (serverName.startsWith("mongodb://")) {
-            // allow mongodb:// URI syntax for the server, to pass everything in one string
-            ret = new MongoClient(new MongoClientURI(serverName, optionsBuilder));
-        } else {
-            ret = new MongoClient(new ServerAddress(serverName), optionsBuilder.build());
+        File configFile = new File(cg.getConfigDir(), CONFIG_NAME);
+        if (configFile.exists()) {
+            MongoDBConnectionConfig config = getDescriptor(configFile, MongoDBConnectionConfig.class);
+            try (MongoClient mongoClient = MongoDBConnectionHelper.newMongoClient(config, builder ->
+                    builder.serverSelectionTimeout((int) TimeUnit.SECONDS.toMillis(getCheckTimeoutInSeconds(cg)))
+                           .description("Nuxeo DB Check"))) {
+                MongoDatabase database = mongoClient.getDatabase(config.dbname);
+                Document ping = new Document("ping", "1");
+                database.runCommand(ping);
+            } catch (MongoTimeoutException e) {
+                throw new ConfigurationException(String.format(
+                        "Unable to connect to MongoDB at %s, please check your connection", config.server));
+            }
         }
-        try {
-            Document ping = new Document("ping", "1");
-            ret.getDatabase(dbName).runCommand(ping);
-        } catch (MongoTimeoutException e) {
-            throw new ConfigurationException(
-                    String.format("Unable to connect to MongoDB at %s, please check your connection", serverName));
-        } finally {
-            ret.close();
+    }
+
+    /**
+     * Creates a descriptor instance for the specified file and descriptor class.
+     * 
+     * @since 11.1
+     */
+    public <T extends Descriptor> T getDescriptor(File file, Class<T> descriptor) throws ConfigurationException {
```

Comment:
I'm not sure that `<T extends Descriptor>` is needed here, just use `<T>`.
Rename `descriptor` -> `klass`